airflow-core/src/airflow/api_fastapi/core_api/routes/ui/dags.py

```diff
@@ -153,6 +156,15 @@ def get_dags(
 
     dags = [dag for dag in session.scalars(dags_select)]
 
+    # Fetch favorite status for each DAG for the current user
+    user_id = str(user.get_id())
+    favorite_dag_ids = set()
+    if dags:
+        favorites_select = select(DagFavorite.dag_id).where(
+            DagFavorite.user_id == user_id, DagFavorite.dag_id.in_([dag.dag_id for dag in dags])
+        )
```

Comment:
We didn't add `if dags` check for existed query that use `[dag.dag_id for dag in dags]`. It seems we can just remove it.

```suggestion

    favorites_select = select(DagFavorite.dag_id).where(
        DagFavorite.user_id == user_id, DagFavorite.dag_id.in_([dag.dag_id for dag in dags])
    )
```